co-simulation/fed/mosaic-infrastructure/src/main/java/org/eclipse/mosaic/fed/infrastructure/ambassador/InfrastructureInstanceManager.java

```diff
@@ -25,20 +25,52 @@
 
 import org.eclipse.mosaic.fed.infrastructure.ambassador.InfrastructureRegistrationMessage;
 import org.eclipse.mosaic.interactions.communication.V2xMessageTransmission;
-import org.eclipse.mosaic.interactions.traffic.VehicleUpdates;
 import org.eclipse.mosaic.lib.enums.AdHocChannel;
-import org.eclipse.mosaic.lib.objects.addressing.AdHocMessageRoutingBuilder;
-import org.eclipse.mosaic.lib.objects.v2x.ExternalV2xContent;
-import org.eclipse.mosaic.lib.objects.v2x.ExternalV2xMessage;
-import org.eclipse.mosaic.lib.objects.v2x.MessageRouting;
-import org.eclipse.mosaic.lib.objects.vehicle.VehicleData;
+import org.eclipse.mosaic.lib.geo.GeoPoint;
 
 /**
  * Session management class for Infrastructure instances communicating with MOSAIC
  * NOTE: TODO See carma.ambassador for reference
  */
 public class InfrastructureInstanceManager {
-    private Map<String, InfrastructureInstance>  managedInstances = new HashMap<>();
+    private Map<String, InfrastructureInstance> managedInstances = new HashMap<>();
     private double currentSimulationTime;
- 
+
+    public void onNewRegistration(InfrastructureRegistrationMessage registration){
+        if (!managedInstances.containsKey(registration.getInfrastructureId())) {
+            try {
+                newInfrastructureInstance(
+                    registration.getInfrastructureId(),
+                    InetAddress.getByName(registration.getRxMessageIpAddress()),
+                    registration.getRxMessagePort(),
+                    registration.getTimeSyncPort(),
+                    registration.getLocation()
+                );
+            } catch (UnknownHostException e) {
+                throw new RuntimeException(e);
```

Comment:
Similar comment about throwing new exception. Either catch exception and log and/or do something here. Or don't catch the exception here and catch it somewhere else and log/do something there.

Reply:
This is modified in this commit: https://github.com/usdot-fhwa-stol/carma-simulation/pull/122/commits/d7b4c932ed70a0faf05d4209dfbf073e26cc956a